File: solutions/elixir/bob/1/lib/bob.ex
defmodule Bob do
  @spec hey(String.t()) :: String.t()
  def hey(input) do
    has_letters? = input =~ ~r/\p{L}/u
    is_upcase? = has_letters? and String.upcase(input) == input
    is_question? = String.last(String.trim(input)) == "?"
    cond do
      input =~ ~r/^\s*$/ -> "Fine. Be that way!"
      is_upcase? and is_question? -> "Calm down, I know what I'm doing!"
      is_upcase? -> "Whoa, chill out!"
      is_question? -> "Sure."
      true -> "Whatever."
    end
  end
end

File: solutions/javascript/bob/1/bob.js
class Bob {
  hey(message) {
    if (message.trim() === '') { return 'Fine. Be that way!'; }
    else if (message.match(/[A-Z]/) && message.toUpperCase() === message) { return 'Whoa, chill out!'; } // add caps-contains regex
    else if (message[message.length - 1] === '?') { return 'Sure.'; }
    return 'Whatever.';
  }
}

export default Bob;